kanidmd/src/lib/credential/mod.rs

```diff
@@ -768,58 +773,40 @@ impl Credential {
         }
     }
 
-    pub(crate) fn update_backup_code(&self, backup_code: HashSet<String>) -> Self {
-        let type_ = match &self.type_ {
-            // TODO: we should panic/reject CredentialType != PasswordMfa?
-            CredentialType::Password(pw) | CredentialType::GeneratedPassword(pw) => {
-                debug_assert!(false);
-                CredentialType::PasswordMfa(
+    pub(crate) fn update_backup_code(
+        &self,
+        backup_codes: BackupCodes, // Should I be passing reference??
```

Comment:
No, because this function will take ownership of the backup_codes so it's correct to NOT use a reference here (if you used a reference you'd need to .clone() )